modules/security-understanding-container-validation.adoc

```diff
@@ -0,0 +1,157 @@
+. For each file that comprises the container image, the cluster downloads the file and calculates its digest.
+
+. The cluster compares the calculated digest of each file that makes up the image with the corresponding digest that is provided in the image manifest.
+If the digest values do not match, the image is rejected.
+
+Once the cluster completes all of the previous steps, the integrity of the image has been validated.
+
+[NOTE]
+====
+The image manifest is validated only when the manifest digest is known, which is true in the following cases:
+
+. The image is pulled by specifying the SHA256 image digest, for example `quay.io/openshift-release-dev/ocp-release@sha256:ea7ac3ad42169b39fce07e5e53403a028644810bee9a212e7456074894df40f3`.
+
+. The image signature is also being validated. See “Image authenticity validation” for more information.
+====
+
+[id="security-understanding-image-authenticity_{context}"]
+== Image authenticity validation
+
+In addition to validating the integrity of images, {product-title} clusters also validate their authenticity by certifying that the images were approved by a trusted source, in this case Red{nbsp}Hat.
+This validation process includes checking the digital signature of the images.
+
+Cluster administrators can use `ClusterImagePolicy` resources to declare image policy for the cluster.
+
+:FeatureName: Configuring cluster image policies
+include::snippets/technology-preview.adoc[]
+
+.Example `ClusterImagePolicy` resource 1
+[source,yaml]
+----
+kind: ClusterImagePolicy
+metadata:
+  name: mypolicy-0
+spec:
+  scopes:
+  - test0.com
+  policy:
+    rootOfTrust:
+      policyType: FulcioCAWithRekor
+      fulcioCAWithRekor:
+        fulcioCAData: dGVzdC1jYS1kYXRhLWRhdGE=
+        fulcioSubject: <example_input_here> <1>
+        rekorKeyData: dGVzdC1yZWtvci1rZXktZGF0YQ==
+    fulcioSubject:
+      oidcIssuer: https://OIDC.example.com
+      signedEmail: test-user@example.com
+    signedIdentity:
+      matchPolicy: RemapIdentity
+      remapIdentity:
+        prefix: test-remap-prefix
+        signedPrefix: test-remap-signed-prefix
+
+----
+<1> Specifies OIDC issuer and the email of the Fulcio authentication configuration.
+
+.Example `ClusterImagePolicy` resource 2
+[source,yaml]
+----
+kind: ClusterImagePolicy
+metadata:
+  name: mypolicy-1
+spec:
+  scopes:
+  - test0.com <1>
+  - test1.com
+  policy:
+    rootOfTrust:
+      policyType: PublicKey
+      publicKey:
+        keyData: dGVzdC1rZXktZGF0YQ==
+----
+<1> This policy for `test0.com` and the policy from `mypolicy-0` will be appended together.
+
+The previous example resources define two policies, both of which apply to `test0.com` pullspecs, and only one of which applies to `test1.com` pullspecs.
+When image integrity is validated at pull time for images whose pullspecs have configured policies with matching `scopes` values, CRI-O will retrieve any Sigstore signature images associated with the image and check them against the configured policies.
```

Comment:
This should probably be a bit more precise about what “matching” means … and I should double-check.
In the underlying c/image, only the _closest_ matching scope is used; I’m not immediately sure wether `{Cluster,}ImagePolicy` also uses the “closest” scope semantics or whether the wider scopes are “inherited” down into the narrower ones.

Reply:
[ClusterImagePolicy API Godocs][1] include:

> If multiple scopes match a given image, only the policy requirements for the most specific scope apply. The policy requirements for more general scopes are ignored.

[1]: https://github.com/openshift/api/blob/07bccfd9266faa9565b51fad64e7a773bfb0f8a6/config/v1alpha1/types_cluster_image_policy.go#L41C5-L42C5

Reply:
What would the wording change look like in light of this feedback? Or do I need to add another sentence to clarify what matching means?

Reply:
I'd just inline the Godocs sentence after the one we're commenting on here, so folks here about how "maching `scopes`..." matter, and then the next sentence gives more precision about what qualifies as "matching".

Reply:
Okay, I added the text snippet you provided